tests/internal/svcproc/svcproc.go

```diff
@@ -4,128 +4,89 @@ import (
 	"context"
 	"errors"
 	"fmt"
-	"os"
 	"os/exec"
-	"path/filepath"
-	"strings"
 	"syscall"
-	"time"
 
 	"go.uber.org/fx"
 
 	"go.autokitteh.dev/autokitteh/backend/svc"
 	"go.autokitteh.dev/autokitteh/sdk/sdkerrors"
 )
 
+const (
+	httpAddrFile = "ak_server_addr.txt"
+)
+
 type svcProc struct {
 	binPath string
-	ropts   svc.RunOptions
 	cfg     *svc.Config
-	addr    string
+	ropts   svc.RunOptions
 	cmd     *exec.Cmd
 	wait    chan svc.ShutdownSignal
 }
 
-func (s *svcProc) Start(ctx context.Context) error {
-	tmpDir, err := os.MkdirTemp("", "ak-svcproc-*")
-	if err != nil {
-		return fmt.Errorf("mkdirtemp: %w", err)
+func NewSvcProc(binPath string, cfg *svc.Config, ropts svc.RunOptions) (svc.Service, error) {
+	if ropts.TemporalClient != nil {
+		return nil, sdkerrors.ErrNotImplemented
 	}
 
-	defer func() {
-		if err := os.RemoveAll(tmpDir); err != nil {
-			fmt.Fprintf(os.Stderr, "rm(%q): %v\n", tmpDir, err)
-		}
-	}()
+	return &svcProc{
+		binPath: binPath,
+		cfg:     cfg,
+		ropts:   ropts,
+	}, nil
+}
+
+func (s *svcProc) Start(ctx context.Context) error {
+	// TODO: Pass user configuration to executable.
+	s.cmd = exec.Command(
+		s.binPath, "up",
+		"--mode", string(s.ropts.Mode),
 
-	addrFilePath := filepath.Join(tmpDir, "addr")
-	readyFilePath := filepath.Join(tmpDir, "ready")
+		"--config", "db.type=sqlite",
+		"--config", "db.dsn=file:autokitteh.sqlite", // In the test's temporary directory.
```

Comment:
plz see https://github.com/autokitteh/autokitteh/pull/232#discussion_r1587134677
I donno what's the right thing to do - either use `sqlite://file:` or just `file:`

Reply:
See my comment in https://linear.app/autokitteh/issue/ENG-769/gormkitteh-db-type-in-dsn

Also note that `sqlite://` is not a valid prefix anyway